kmod/src/dir.c

```diff
@@ -1111,46 +1111,55 @@ static void *scoutfs_follow_link(struct dentry *dentry, struct nameidata *nd)
 				   corrupt_symlink_missing_item,
 				   "ino %llu size %llu", scoutfs_ino(inode),
 				   size);
+		kfree(path);
 		ret = -EIO;
 
 	} else if (ret == 0 && path[size - 1]) {
 		scoutfs_corruption(sb, SC_SYMLINK_NOT_NULL_TERM,
 				   corrupt_symlink_not_null_term,
 				   "ino %llu last %u",
 				   scoutfs_ino(inode), path[size - 1]);
+		kfree(path);
 		ret = -EIO;
 	}
 
 out:
-	if (ret < 0) {
-		kfree(path);
+	if (ret < 0)
 		path = ERR_PTR(ret);
-	} else {
-		nd_set_link(nd, path);
-	}
+
 	scoutfs_unlock(sb, inode_lock, SCOUTFS_LOCK_READ);
 	return path;
 }
 
+#ifdef KC_LINUX_HAVE_RHEL_IOPS_WRAPPER
+static void *scoutfs_follow_link(struct dentry *dentry, struct nameidata *nd)
+{
+	char *path;
+
+	path = scoutfs_get_link_target(dentry);
+	if (!IS_ERR_OR_NULL(path))
+		nd_set_link(nd, path);
+	return NULL;
```

Comment:
This needs to return path.  Returning NULL masks errors and leaks that path.  It's the return from this which is passed to put_link to be freed.